packages/design-system-react-native/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,234 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, fireEvent } from '@testing-library/react-native';
+import type { ReactElement } from 'react';
+import React, { createRef } from 'react';
+import { Text } from 'react-native';
+import type { StyleProp, ViewStyle } from 'react-native';
+
+import { Checkbox } from './Checkbox';
+
+/**
+ * Helper to compute expected container styles for the checkbox.
+ *
+ * @param tw - The tailwind instance.
+ * @param options - Checkbox state options.
+ * @param options.selected - Whether the checkbox is selected.
+ * @param options.invalid - Whether the checkbox is invalid.
+ * @param options.pressed - Whether the checkbox is pressed.
+ * @returns The expected container style.
+ */
+function getExpectedContainerStyle(
+  tw: ReturnType<typeof useTailwind>,
+  options: { selected?: boolean; invalid?: boolean; pressed?: boolean },
+): StyleProp<ViewStyle> {
+  const { selected = false, invalid = false, pressed = false } = options;
+  const baseBg = selected ? 'bg-primary-default' : 'bg-background-default';
+  let baseBorder = 'border-border-default';
+  if (selected) {
+    baseBorder = 'border-primary-default';
+  } else if (invalid) {
+    baseBorder = 'border-error-default';
+  }
+  const pressedBg = selected
+    ? 'bg-primary-defaultPressed'
+    : 'bg-background-defaultPressed';
+  let pressedBorder = 'border-border-default';
+  if (selected) {
+    pressedBorder = 'border-primary-defaultPressed';
+  } else if (invalid) {
+    pressedBorder = 'border-error-default';
+  }
+  const stateClasses = pressed
+    ? `${pressedBg} ${pressedBorder}`
+    : `${baseBg} ${baseBorder}`;
+  return tw`${stateClasses} flex h-[22px] w-[22px] items-center justify-center rounded border-2`;
+}
```

Comment:
I know we are trying to assert that the generated styles match the tailwind classname inputs that we are using with twrnc but I still have some concerns with this approach.

- We're duplicating the style logic from the component in the test, which means if the style logic changes, we must update both the code and the test.
- This doesn't truly verify the rendered output, but rather checks if two separately generated objects (using the same logic) happen to match.
- We're also tightly coupling our tests to the implementation details of our style generation.

I wonder if snapshot testing may be a better approach. I've posted in [twrnc discussions for any tips](https://github.com/jaredh159/tailwind-react-native-classnames/discussions/346)

Reply:
This is to just easily grab the right class name. I can remove this and hardcode the class name instead. wdyt?

Reply:
Yes, I think so, we want to reduce the amount of logic we are writing in our tests.